src/java.base/share/classes/java/lang/Process.java

```diff
@@ -159,9 +195,14 @@ public Process() {}
      * then the input stream returned by this method will receive the
      * merged standard output and the standard error of the process.
      *
+     * <p>The input stream should be {@linkplain InputStream#close closed}
+     * when it is no longer needed.
+     *
      * @apiNote
-     * Use {@link #getInputStream()} and {@link #inputReader()} with extreme care.
-     * The {@code BufferedReader} may have buffered input from the input stream.
+     * Avoid using both {@link #getInputStream} and {@link #inputReader(Charset)}.
```

Comment:
The reworded sentence I think is still a bit confusing. In the new proposed form, it sounds like both these methods are to be avoided. I think the intention of this apiNote is to convey that they should not be invoked on the same Process instance. Would something like the following be clear and accurate:

>
> Either use {@link #getInputStream} or {@link #inputReader(Charset)} but not both on the same {@code Process}.
> ...

Reply:
Same comment about the new text on `getErrorStream()`

Reply:
Also, I suspect the use of `{@link #inputReader(Charset)}` instead of the pre-existing `{@link #inputReader()}` was perhaps unintentional?